### ISSUE TITLE
Question on importing plugin dependencies

### ISSUE THREAD

**Issue by MeoMix** [jspm/registry#565]
Hey,

I wanted to use the following plugin: https://github.com/davis/plugin-hbs

It's a plugin which has a declared dependency on Handlebars.

When I installed it through `jspm install hbs` my config file installed a dependency on handlebars:

    "hbs": "github:davis/plugin-hbs@1.1.0",
    "github:davis/plugin-hbs@1.1.0": {
      "handlebars": "github:components/handlebars.js@4.0.2"
    },

However, I would still encounter an error saying that handlebars was not defined when attempting to use the plugin.

The solution was to manually install handlebars to add this entry to my config:

    "handlebars": "github:components/handlebars.js@4.0.2",

That doesn't feel right to me.

Does this plugin have its dependencies misconfigured? How can they be corrected?

**Comment by guybedford** [comment]
Yes this is a misconfiguration in the plugin. Typically plugins need access to helpers from the handlebars compiler, so will output templates that request the helpers.

Plugin authors need to make sure they don't output compiled templates that do `require('hbs')` but rather output templates that do `System.normalize('hbs', module.id)` for CJS or `System.normalize('hbs', __moduleName)`. That is the plugin should first determine the normalized path to the runtime before substituting that path into the `require('normalized/path/to/hbs')`.

**Comment by MeoMix** [comment]
@davis Could you take a look at fixing this? :)

**Comment by davis** [comment]
ah got it, thanks!
